### PR TITLE
Add convenience methods to `WSConnectionHighLevel`

### PR DESCRIPTION
**Pull request by armanbilge**
I'm also sneaking in a couple convenience methods to the high-level client.

### CODE REVIEW

File: client/src/main/scala/org/http4s/client/websocket/WSClient.scala [hunk 1]
```diff
@@ -130,7 +130,13 @@ private[http4s] trait WSConnection[F[_]] { outer =>
...
 }
 
-private[http4s] trait WSConnectionHighLevel[F[_]] { outer =>
+trait WSConnectionHighLevel[F[_]] { outer =>
+
+  /** Send a single websocket text frame. The sending side of this connection has to be open. */
+  def send(text: String): F[Unit] = send(WSFrame.Text(text))
+
+  /** Send a single websocket binary frame. The sending side of this connection has to be open. */
+  def send(bytes: ByteVector): F[Unit] = send(WSFrame.Binary(bytes))
```
Comment (armanbilge) [inline client/src/main/scala/org/http4s/client/websocket/WSClient.scala:139]: Thoughts on overloads vs `sendText` and `sendBytes`?

**Comment by armanbilge** [comment]
Looks like the jdk client is not going to implement this for 0.22, see https://github.com/http4s/http4s-jdk-http-client/issues/568.

http4s-dom also doesn't support 0.22 at all which doesn't support JS at all.

Which leads me to wonder if we should make this public at all? Since we have no concrete plans to implement it at this time.

File: client/src/main/scala/org/http4s/client/websocket/WSClient.scala (hunk 1 above)
Comment (rossabaker) [reply client/src/main/scala/org/http4s/client/websocket/WSClient.scala:139]: I tend to avoid overloads these days.

Comment (armanbilge) [reply client/src/main/scala/org/http4s/client/websocket/WSClient.scala:139]: Yeah, it's good practice. But I had to try :P

**Comment by rossabaker** [comment]
I would not make it public on this branch before there is at least one, preferably two, concrete implementations.  If one drops from the sky, we can make it public here, but it's probably better to unveil it in 0.23.

**Review by rossabaker — APPROVED** [review]

**Pull request merged**